amqphandler/amqphandler.go

```diff
@@ -612,31 +624,44 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
 	}
 }
 
-func (handler *AmqpHandler) decodeData(data []byte, result interface{}) error {
+func (handler *AmqpHandler) unmarshalReceiveData(data []byte) (Message, error) {
 	if len(data) == 0 {
-		return nil
+		return nil, nil
 	}
 
-	decryptData, err := handler.cryptoContext.DecryptMetadata(data)
-	if err != nil {
-		return aoserrors.Wrap(err)
+	// unmarshal type name
+	var messageType struct {
+		Type string `json:"messageType"`
 	}
 
-	if err = json.Unmarshal(decryptData, result); err != nil {
-		return aoserrors.Wrap(err)
+	if err := json.Unmarshal(data, &messageType); err != nil {
+		return nil, aoserrors.New("JSON parsing problem")
```

Comment:
no need to override err, return what unmarshal returns

Reply:
fixed